nano/node/socket.cpp

```diff
@@ -512,3 +512,19 @@ void nano::server_socket::evict_dead_connections ()
 		++it;
 	}
 }
+
+std::string nano::socket_type_to_string (nano::socket::type_t type)
+{
+	switch (type)
+	{
+		case nano::socket::type_t::undefined:
```

Comment:
For consistency this should be renamed to handshake rather than undefined since it's referenced by is_handshake_connection.

Reply:
But calling it handshake would be incorrect for bootstrap connections. Wouldn't it?
We could possibly have an additional state calling handshake when we have started a handshake but before completing it.

Reply:
I think `is_handshake_connection()` should be renamed to `is_undefined_connection()` because that is exactly what it is checking.

Reply:
Could be renamed, it's true that there won't be any handshake if the connection is used for bootstrapping. I would move the whole socket type tracking into bootstrap_server (and rename that to network_server) because at the end of the day socket should be oblivious as to what it's used for, this is the task for higher layers